yeat/tests/test_cli.py

```diff
@@ -164,3 +167,25 @@ def test_custom_coverage_input(coverage, capsys, tmp_path):
     assert df.iloc[12]["sample_contigs"] == 56  # num_contigs
     assert df.iloc[13]["sample_contigs"] == 35168  # largest_contig
     assert df.iloc[14]["sample_contigs"] == 199940  # total_len
+
+
+@pytest.mark.long
+@pytest.mark.parametrize("execution_number", range(3))
+def test_random_downsample_seed(execution_number, capsys, tmp_path):
+    wd = str(tmp_path)
+    arglist = [
+        data_file("megahit.cfg"),
+        data_file("short_reads_1.fastq.gz"),
+        data_file("short_reads_2.fastq.gz"),
+        "--outdir",
+        wd,
+        "-d",
+        "2000",
+    ]
+    args = cli.get_parser().parse_args(arglist)
+    cli.main(args)
+    quast_report = Path(wd).resolve() / "analysis" / "quast" / "megahit" / "report.tsv"
+    df = pd.read_csv(quast_report, sep="\t")
+    assert 61 <= df.iloc[12]["sample_contigs"] <= 91  # 76 +-20% of avg num_contigs
+    assert 4183 <= df.iloc[13]["sample_contigs"] <= 6273  # 5228 +-20% of avg largest_contig
+    assert 59515 <= df.iloc[14]["sample_contigs"] <= 89271  # 74393 +-20% of avg total_len
```

Comment:
https://github.com/bioforensics/yeat/issues/27#issuecomment-1301009696

This is my take on this suggestion.

I was pretty liberal on my +- buffer range.

The way I determined my medium for each assert is:
1. I ran the above list of arguments 5 times, 
2. took the average for `num_contigs`, `largest_contig`, and `total_len` and 
3. calculated the buffer +- caps.

Above the function, there is a decorator. When this function is executed with pytest, the function is called 3 times. Since, the seed is random by default, we do not need to specify the seed.

Reply:
Looks good! A couple comments.

I have no idea what information `df.iloc[12]["sample_contigs"]` stores. There *is* a describing its contents at the end of the line, but comments have a habit of coming out of sync with the code they are intended to describe. Probably better for legibility and clarity to assign those values to descriptive variable names before the assertion tests.

I think the `x <= var <= y` is a clear construction, but another you may consider uses `pytest.approx`. I use this most frequently to test the value of floating point numbers, for which simple `==` equality tests often fail (even if you're looking for an "exact" value, you have to specify some level of tolerance). But you can apply the same idea here, and just specify a wide tolerance. The first line would then become something like this, which is a pretty clear representation of 76 +/- 15.

```python
assert num_contigs == pytest.approx(76, abs=15)
```